server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -438,58 +435,57 @@ private void loadAllLookupsAndInitStateRef()
-        )
+    ExecutorService executorService = Execs.multiThreaded(
+        lookupConfig.getNumLookupLoadingThreads(),
+        "LookupReferencesManager-Startup-%s"
     );
+    ExecutorCompletionService completionService = new ExecutorCompletionService(executorService);
     try {
-      List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
       LOG.info("Starting lookup loading process");
       for (LookupBean lookupBean : lookupBeanList) {
-        futures.add(
-            executorService.submit(
-                () -> {
-
-                  LookupExtractorFactoryContainer container = lookupBean.getContainer();
-                  LOG.info(
-                      "Starting lookup [%s]:[%s]",
-                      lookupBean.getName(),
-                      container
-                  );
-                  if (container.getLookupExtractorFactory().start()) {
-                    LOG.info(
-                        "Started lookup [%s]:[%s]",
-                        lookupBean.getName(),
-                        container
-                    );
-                    return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
-                  } else {
-                    LOG.error(
-                        "Failed to start lookup [%s]:[%s]",
-                        lookupBean.getName(),
-                        container
-                    );
-                    return null;
-                  }
-                }
-            )
-        );
-      }
-      final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
-      try {
-        futureList.get()
-                  .stream()
-                  .filter(Objects::nonNull)
-                  .forEach(builder::put);
+        completionService.submit(
+            () -> {
 
-        stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
-      }
-      catch (InterruptedException | ExecutionException ex) {
-        futureList.cancel(true);
-        throw ex;
+              LookupExtractorFactoryContainer container = lookupBean.getContainer();
+              LOG.info(
+                  "Starting lookup [%s]:[%s]",
+                  lookupBean.getName(),
+                  container
+              );
+              if (container.getLookupExtractorFactory().start()) {
+                LOG.info(
+                    "Started lookup [%s]:[%s]",
+                    lookupBean.getName(),
+                    container
+                );
+                return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+              } else {
+                LOG.error(
+                    "Failed to start lookup [%s]:[%s]",
+                    lookupBean.getName(),
+                    container
+                );
+                return null;
+              }
+            }
+        );
       }
+      IntStream.range(0, lookupBeanList.size())
```

Comment:
Please just use loop, also it will make easier to fall thorough with InterruptedException below.